### PR TITLE
CNV-16317: RN nmstate not supported by CNV

### PR DESCRIPTION
**Pull request by sjhala-ccs**
Version(s): 4.11

Issue: [CNV-16317](https://issues.redhat.com//browse/CNV-16317)

Link to docs preview: http://file.rdu.redhat.com/sjhala/cnv-16317/virt/virt-4-11-release-notes.html#virt-4-11-removed

### CODE REVIEW

File: virt/virt-4-11-release-notes.adoc [hunk 1]
```diff
@@ -105,6 +105,7 @@ Deprecated features are included in the current release and supported. However,
 Removed features are not supported in the current release.
 
 //CNV-16317 NMState is no longer part of CNV
+* {VirtProductName} {VirtVersion} removes support for the link:https://nmstate.io/[nmstate] library. To continue using `nmstate` features, you must install the xref:../networking/k8s_nmstate/k8s-nmstate-about-the-k8s-nmstate-operator.adoc#k8s-nmstate-about-the-k8s-nmstate-operator[Kubernetes NMState Operator] from the *OperatorHub* in the {product-title} web console or by using the OpenShift CLI (`oc`) before updating {VirtProductName}.
```
Comment (phoracek) [inline virt/virt-4-11-release-notes.adoc:108]:
```suggestion
* {VirtProductName} {VirtVersion} removes support for the link:https://nmstate.io/[nmstate] (Node Network State, Node Network Configuration Policy, and Node Network Configuration Enactment APIs). To continue using nmstate features, you must install the xref:../networking/k8s_nmstate/k8s-nmstate-about-the-k8s-nmstate-operator.adoc#k8s-nmstate-about-the-k8s-nmstate-operator[Kubernetes NMState Operator] from the *OperatorHub* in the {product-title} web console or by using the OpenShift CLI (`oc`) before updating {VirtProductName}.
```

**Review by phoracek — CHANGES_REQUESTED** [review]

File: virt/virt-4-11-release-notes.adoc [hunk 2]
```diff
@@ -105,6 +105,13 @@ Deprecated features are included in the current release and supported. However,
 Removed features are not supported in the current release.
 
 //CNV-16317 NMState is no longer part of CNV
+* {VirtProductName} {VirtVersion} removes support for the link:https://nmstate.io/[nmstate] library including the following objects:
```
Comment (phoracek) [inline virt/virt-4-11-release-notes.adoc:108]: nmstate is not a library (at least not the way we use it)

```suggestion
* {VirtProductName} {VirtVersion} removes support for the link:https://nmstate.io/[nmstate] including the following objects:
```

File: virt/virt-4-11-release-notes.adoc [hunk 3]
```diff
@@ -105,6 +105,13 @@ Deprecated features are included in the current release and supported. However,
...
++
+** `NodeNetworkState`
+** `NodeNetworkConfigurationPolicy`
+** `NodeNetworkConfigurationEnactment`
++
+To continue using `nmstate` features, you must first install the xref:../networking/k8s_nmstate/k8s-nmstate-about-the-k8s-nmstate-operator.adoc#k8s-nmstate-about-the-k8s-nmstate-operator[Kubernetes NMState Operator] from the *OperatorHub* in the {product-title} web console, or by using the OpenShift CLI (`oc`), before updating to {VirtProductName} {VirtVersion}.
```
Comment (phoracek) [inline virt/virt-4-11-release-notes.adoc:114]: I would not use `quotes` for nmstate. It is not an executable or field in a yaml

Comment (phoracek) [inline virt/virt-4-11-release-notes.adoc:114]: Can we add that "All existing configuration will be preserved and will remain supported by the standalone Kubernetes NMState  Operator."

File: virt/virt-4-11-release-notes.adoc [hunk 4]
```diff
@@ -105,6 +105,15 @@ Deprecated features are included in the current release and supported. However,
 Removed features are not supported in the current release.
 
 //CNV-16317 NMState is no longer part of CNV
+* {VirtProductName} {VirtVersion} removes support for link:https://nmstate.io/[nmstate] including the following objects:
```
Comment (ousleyp) [inline virt/virt-4-11-release-notes.adoc:108]: Are the objects examples of objects that are not supported, or are they the only objects not supported? If they're examples (and all nmstate objects are no longer supported), there should be a comma after `[nmstate]`.

File: virt/virt-4-11-release-notes.adoc [hunk 5]
```diff
@@ -105,6 +105,15 @@ Deprecated features are included in the current release and supported. However,
...
++
+--
+** `NodeNetworkState`
+** `NodeNetworkConfigurationPolicy`
+** `NodeNetworkConfigurationEnactment`
+--
++
+To continue using nmstate features, you must first install the xref:../networking/k8s_nmstate/k8s-nmstate-about-the-k8s-nmstate-operator.adoc#k8s-nmstate-about-the-k8s-nmstate-operator[Kubernetes NMState Operator] from the *OperatorHub* in the {product-title} web console, or by using the OpenShift CLI (`oc`), before updating to {VirtProductName} {VirtVersion}. The Kubernetes NMState Operator preserves and supports the existing nmstate configuration.
```
Comment (ousleyp) [inline virt/virt-4-11-release-notes.adoc:116]: If nmstate is no longer supported, do clusters become unsupported if they use this workaround? Or is nmstate still supported, but no longer being provided by CNV?

I think the support implications here should be clearer.

Comment (ousleyp) [inline virt/virt-4-11-release-notes.adoc:116]: Do I understand correctly that regardless of the method of (NMState Operator) installation, the NMState Operator must be installed _before_ the cluster is updated to 4.11? I think the first sentence is a bit ambiguous and might benefit from being broken into two sentences. 

Suggestion:
```
... you must install the ... [Kubernetes NMState Operator] before updating to {VirtProductName} {VirtVersion}. You can install it from the *OperatorHub* in the {product-title} web console or by using the OpenShift CLI (`oc`). The Kubernetes NMState Operator preserves ...
```

**Review by ousleyp — COMMENTED** [review]
This is looking good, but I have a couple of questions/suggestions. Thanks, Shikha!

File: virt/virt-4-11-release-notes.adoc (hunk 5 above)
Comment (sjhala-ccs) [reply virt/virt-4-11-release-notes.adoc:116]: As far as I understand, nmstate is no longer being provided by CNV, but still supported by OCP through the Kubernetes NMState Operator.

Comment (yossisegev) [reply virt/virt-4-11-release-notes.adoc:116]: > As far as I understand, nmstate is no longer being provided by CNV, but still supported by OCP through the Kubernetes NMState Operator.

Exactly

File: virt/virt-4-11-release-notes.adoc [hunk 6]
```diff
@@ -105,6 +105,15 @@ Deprecated features are included in the current release and supported. However,
...
 //CNV-16317 NMState is no longer part of CNV
+* {VirtProductName} {VirtVersion} removes support for link:https://nmstate.io/[nmstate], including the following objects:
++
+--
+** `NodeNetworkState`
+** `NodeNetworkConfigurationPolicy`
+** `NodeNetworkConfigurationEnactment`
+--
++
+Instead, you can use nmstate features by installing the xref:../networking/k8s_nmstate/k8s-nmstate-about-the-k8s-nmstate-operator.adoc#k8s-nmstate-about-the-k8s-nmstate-operator[Kubernetes NMState Operator] before updating to {VirtProductName} {VirtVersion}. You can install it from the *OperatorHub* in the {product-title} web console, or by using the OpenShift CLI (`oc`). The Kubernetes NMState Operator preserves and supports the existing nmstate configuration.
```
Comment (ousleyp) [inline virt/virt-4-11-release-notes.adoc:116]: I wonder if users will find the phrase "Instead, you can use nmstate features by..." a bit unclear re: support implications. What about something like this?

```suggestion
To preserve and support your existing nmstate configuration, install the xref:../networking/k8s_nmstate/k8s-nmstate-about-the-k8s-nmstate-operator.adoc#k8s-nmstate-about-the-k8s-nmstate-operator[Kubernetes NMState Operator] before updating to {VirtProductName} {VirtVersion}. You can install it from the *OperatorHub* in the {product-title} web console, or by using the OpenShift CLI (`oc`). 
```

**Review by ousleyp — COMMENTED** [review]
LGTM; just one suggestion to consider. Thanks!

**Pull request merged**